### PR TITLE
Add virtio-win viorng driver load/unload test

### PR DESCRIPTION
**Pull request by suqinhuang**
Load/unload viorng driver befor/during/after rng_bat test

Signed-off-by: Suqin Huang shuang@redhat.com

### CODE REVIEW

**Comment by suqinhuang** [comment]
id: 1231092

**Comment by suqinhuang** [comment]
https://github.com/avocado-framework/avocado-vt/pull/585

File: qemu/tests/cfg/driver_load_stress.cfg [hunk 1]
```diff
@@ -106,3 +106,16 @@
...
             post_cmd += "reg add HKCU\Software\Microsoft\MediaPlayer\Preferences /v ModeLoop /t REG_DWORD /d 0 /f"
             target_process = wmplayer.exe
+        - with_rng:
+            driver_name = viorng
+            run_bgstress = rng_bat
+            target_process = D:\random_%PROCESSOR_ARCHITECTURE%.exe
+            session_cmd_timeout = 240
+            read_rng_cmd  = "X:\random_%PROCESSOR_ARCHITECTURE%.exe"
+            rng_data_rex = "0x\w"
+            driver_id_pattern = "(.*?):.*?VirtIO RNG Device"
```
Comment (CongLi) [inline qemu/tests/cfg/driver_load_stress.cfg]: I think also need update driver_id_cmd.
driver_id_cmd = C:\devcon.exe find * | find "VirtIO"

If some drive if not installed, test will be failed.

17:40:03 ERROR| TestFail: Didn't find driver info from guest PCI\VEN_1AF4&DEV_1000&SUBSYS_00011AF4&REV_00\3&13C0B0C5&0&28: Red Hat VirtIO Ethernet Adapter
17:40:03 ERROR| PCI\VEN_1AF4&DEV_1001&SUBSYS_00021AF4&REV_00\3&13C0B0C5&0&20: Red Hat VirtIO SCSI controller
17:40:03 ERROR| SCSI\DISK&VEN_RED_HAT&PROD_VIRTIO\4&35110308&0&000000       : Red Hat VirtIO SCSI Disk Device

**Comment by suqinhuang** [comment]
https://github.com/avocado-framework/avocado-vt/pull/642/

**Comment by suqinhuang** [comment]
1-smp_1.4096m.repeat1.run_test.Host_RHEL.m7.u2.rng_random.qcow2.virtio_blk.up.virtio_net.Win7.x86_64.sp1.io-github-autotest-qemu.driver_load_stress.with_rng.before_bg_test

**Comment by CongLi** [comment]
read_rng_cmd  = "X:\random_%PROCESSOR_ARCHITECTURE%.exe" should use a loop for during test.

**Comment by suqinhuang** [comment]
PASS 1-smp_1.4096m.repeat1.run_test.Host_RHEL.m7.u2.rng_random.qcow2.virtio_blk.up.virtio_net.Win7.x86_64.sp1.io-github-autotest-qemu.driver_load_stress.before_bg_test.with_viorng
2-smp_1.4096m.repeat1.run_test.Host_RHEL.m7.u2.rng_random.qcow2.virtio_blk.up.virtio_net.Win7.x86_64.sp1.io-github-autotest-qemu.driver_load_stress.during_bg_test.with_viorng
3-smp_1.4096m.repeat1.run_test.Host_RHEL.m7.u2.rng_random.qcow2.virtio_blk.up.virtio_net.Win7.x86_64.sp1.io-github-autotest-qemu.driver_load_stress.after_bg_test.with_viorng

**Comment by CongLi** [comment]
Met error in session log:
2016-08-16 15:33:59: C:\>"D:\random_x86.exe"
2016-08-16 15:33:59: **** Error 0xc0000135 returned by BCryptOpenAlgorithmProvider

15:41:20 INFO | PASS 1-smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.rng_random.qcow2.virtio_blk.up.virtio_net.Win2012.x86_64.r2.io-github-autotest-qemu.driver_load_stress.during_bg_test.with_viorng

And there is no random number generated, but test pass.

**Comment by pingl2016** [comment]
Case driver_load_stress.before_bg_test.with_viorng and driver_load_stress.after_bg_test.with_viorng
failed with log below. 
04:17:30 ERROR| Reproduced traceback from: /usr/lib/python2.7/site-packages/avocado_plugins_vt-39.0-py2.7.egg/avocado_vt/test.py:420
04:17:30 ERROR| Traceback (most recent call last):
04:17:30 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-39.0-py2.7.egg/avocado_vt/test.py", line 196, in runTest
04:17:30 ERROR|     self._runTest()
04:17:30 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-39.0-py2.7.egg/avocado_vt/test.py", line 337, in _runTest
04:17:30 ERROR|     run_func(self, params, env)
04:17:30 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-39.0-py2.7.egg/virttest/error_context.py", line 135, in new_fn
04:17:30 ERROR|     return fn(*args, **kwargs)
04:17:30 ERROR|   File "/usr/share/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/driver_in_use.py", line 104, in run
04:17:30 ERROR|     run_subtest(bg_stress_test)
04:17:30 ERROR|   File "/usr/share/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/driver_in_use.py", line 80, in run_subtest
04:17:30 ERROR|     utils_test.run_virt_sub_test(test, params, env, sub_type)
04:17:30 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-39.0-py2.7.egg/virttest/utils_test/__init__.py", line 1242, in run_virt_sub_test
04:17:30 ERROR|     run_func(test, params, env)
04:17:30 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-39.0-py2.7.egg/virttest/error_context.py", line 135, in new_fn
04:17:30 ERROR|     return fn(*args, **kwargs)
04:17:30 ERROR|   File "/usr/share/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/rng_bat.py", line 100, in run
04:17:30 ERROR|     "guest: %s" % output)
04:17:30 ERROR| TestFail: Unable to read random numbers fromguest: **** Error 0xc0000135 returned by BCryptOpenAlgorithmProvider

**Comment by pingl2016** [comment]
After apply the patch below, cases will pass.
# git diff  shared/cfg/guest-os/Windows.cfg 
diff --git a/shared/cfg/guest-os/Windows.cfg b/shared/cfg/guest-os/Windows.cfg
index d85ebd7..a37c907 100644
--- a/shared/cfg/guest-os/Windows.cfg
+++ b/shared/cfg/guest-os/Windows.cfg
@@ -263,7 +263,7 @@
         copy_from_command = copy %s\notepad.exe c:\ /y
         compare_command = fc /b c:\windows\system32\notepad.exe c:\notepad.exe
         check_result_key_word = no difference
-    driver_load_stress.with_rng, rng_bat, viorng_in_use:
+    driver_load_stress.before_bg_test.with_viorng, driver_load_stress.after_bg_test.with_viorng, rng_bat, viorng_in_use:
         shell_client = telnet
         shell_port = 23
     floppy_test:

**Comment by suqinhuang** [comment]
https://github.com/avocado-framework/avocado-vt/pull/660/files

**Comment by pingl2016** [comment]
The result of case driver_load_stress.during_bg_test.with_viorng is unstable. Sometimes it failed as  "the limit on number of connections"
04:17:27 DEBUG| Client said 'connection closed'    (output: '\nDenying new connections due to the limit on number of connections.\x00Connection closed by foreign host.\n')
04:17:29 DEBUG| Client said 'connection refused'    (output: 'Trying 10.66.145.132...\ntelnet: connect to address 10.66.145.132: Connection refused\n')
04:17:29 DEBUG| Exception in thread Thread-12:
04:17:29 DEBUG| Traceback (most recent call last):
04:17:29 DEBUG|   File "/usr/lib64/python2.7/threading.py", line 811, in __bootstrap_inner
04:17:29 DEBUG|     self.run()
04:17:29 DEBUG|   File "/usr/lib/python2.7/site-packages/autotest/client/shared/utils.py", line 979, in run
04:17:29 DEBUG|     self._retval = self._target(*self._args, **self._kwargs)
04:17:29 DEBUG|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-39.0-py2.7.egg/virttest/utils_test/__init__.py", line 1242, in run_virt_sub_test
04:17:29 DEBUG|     run_func(test, params, env)
04:17:29 DEBUG|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-39.0-py2.7.egg/virttest/error_context.py", line 135, in new_fn
04:17:29 DEBUG|     return fn(*args, **kwargs)
04:17:29 DEBUG|   File "/usr/share/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/rng_bat.py", line 97, in run
04:17:29 DEBUG|     output = session.cmd_output(read_rng_cmd, timeout=read_rng_timeout)
04:17:29 DEBUG|   File "/usr/lib/python2.7/site-packages/aexpect/client.py", line 1059, in cmd_output
04:17:29 DEBUG|     raise ShellProcessTerminatedError(cmd, e.status, o)
04:17:29 DEBUG| ShellProcessTerminatedError: Shell process terminated while waiting for command to complete: 'for /l %i in (1, 1, 1000) do "D:\\random_%PROCESSOR_ARCHITECTURE%.exe"'    (status: 1,    output: 'Generated random number is\n0xc6, 0xdd, 0x35, 0xc8, 0x94, 0x4a, 0x2d, 0x67, 0xcd, 0x04, \n0xf2, 0xb2, 0xdc, 0xc8, 0x8c, 0xee, 0x45, 0xda, 0x5f, 0xba, \n0x95, 0xfc, 0xc9, 0x9f, 0x48, 0xae, 0xaa, 0xbe, 0x1a, 0x39, \n0x5d, 0xb5, 0x8c, 0x52, 0xda, 0x70, 0x60, 0x0b, 0xf0, 0x87, \n0xfc, 0x80, 0xb9, 0xcd, 0xfe, 0x37, 0x01, 0xaa, 0x50, 0x3a, \n0xae, 0xac, 0xe3, 0x5a, 0xb2, 0x2b, 0x16, 0x7a, 0x90, 0xc7, \n0xe0, 0x10, 0x43, 0x6e, 0x8a, 0x18, 0x9f, 0xf8, 0xdb, 0xd3, \n0x09, 0x49, 0x0a, 0x0c, 0xec, 0x25, 0x02, 0x5a, 0x36, 0x5f, \n0x41, 0x17, 0x3a, 0xbe, 0x49, 0x16, 0x8e, 0x08, 0x14, 0xb2, \n0xe8, 0xb2, 0xc8, 0x42, 0x33, 0xee, 0x4c, 0xd1, 0x7d, 0x31, \n0xa7, 0xac, 0xb5, 0xfa, 0xcc, 0x03, 0x86, 0xd0, 0x71, 0xb9, \n0xbd, 0x7a, 0x29, 0xf3, 0x1b, 0x9a, 0xd4, 0xfa, 0x86, 0xa1, \n0x69, 0x77, 0x1c, 0xba, 0x7c, 0x23, 0x6e, 0x56, Generated random number is\n0xad, 0xd8, 0xbd, 0x4d, 0x5e, 0xda, 0x87, 0x96, 0xbe, 0xc2, \n0x4b, 0x47, 0x4b, 0x7d, 0xc6, 0xca, 0x8d, 0x7a, 0x13, 0x26, \n0xb6, 0x7e, 0x2d, 0xb5, 0x3e, 0xfe, 0xaf, 0x35, 0x96, 0x21, \n0x26, 0x86, 0x25, 0x3b, 0x39, 0xb5, 0x03, 0x01, 0xba, 0x43, \n0x6b, 0x36, 0xfd, 0x0b, 0x86, 0x1f, 0x29, 0x0a, 0x2a, 0x97, \n0x10, 0xba, 0x05, 0x9c, 0x43, 0xd0, 0x2f, 0x69, 0xa5, 0xc0, \n0x8c, 0xa5, 0xe2, 0x8b, 0x1d, 0x16, 0xfc, 0x84, 0x37, 0xb8, \n0x95, 0x79, 0x01, 0xf2, 0xec, 0xa6, 0x04, 0x64, 0x30, 0x5f, \n0x19, 0xd2, 0xdc, 0x5f, 0xca, 0xdc, 0xb2, 0x62, 0x88, 0xa9, \n0x63, 0xe0, 0xcb, 0x28, 0x8a, 0x71, 0x2c, 0x23, 0xf4, 0x88, \n0xc6, 0xb0, 0x2c, 0xd1, 0x30, 0x2b, 0xaf, 0x78, 0x6a, 0x76, \n0x4b, 0x5a, 0xbf, 0x60, 0x3f, 0x52, 0xee, 0xf0, 0x51, 0x0d, \n0x38, 0x3b, 0xd2, 0x14, 0xc6, 0x88, 0x48, 0x1a, \nmessage from the administrator at WIN-3I1Q7LHRQ4O on 8/17/2016 4:17:27 AM\n\x00\nThe computer is shutting down......\n\x00Connection closed by foreign host.\n')
04:18:45 ERROR| Reproduced traceback from: /usr/lib/python2.7/site-packages/avocado_plugins_vt-39.0-py2.7.egg/avocado_vt/test.py:420
04:18:45 ERROR| Traceback (most recent call last):
04:18:45 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-39.0-py2.7.egg/avocado_vt/test.py", line 196, in runTest
04:18:45 ERROR|     self._runTest()
04:18:45 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-39.0-py2.7.egg/avocado_vt/test.py", line 337, in _runTest
04:18:45 ERROR|     run_func(self, params, env)
04:18:45 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-39.0-py2.7.egg/virttest/error_context.py", line 135, in new_fn
04:18:45 ERROR|     return fn(*args, **kwargs)
04:18:45 ERROR|   File "/usr/share/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/driver_in_use.py", line 110, in run
04:18:45 ERROR|     run_subtest(sub_type)
04:18:45 ERROR|   File "/usr/share/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/driver_in_use.py", line 80, in run_subtest
04:18:45 ERROR|     utils_test.run_virt_sub_test(test, params, env, sub_type)
04:18:45 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-39.0-py2.7.egg/virttest/utils_test/__init__.py", line 1242, in run_virt_sub_test
04:18:45 ERROR|     run_func(test, params, env)
04:18:45 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-39.0-py2.7.egg/virttest/error_context.py", line 135, in new_fn
04:18:45 ERROR|     return fn(*args, **kwargs)
04:18:45 ERROR|   File "/usr/share/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/driver_load.py", line 101, in run
04:18:45 ERROR|     load_driver(session, driver_load_cmd, driver_id)
04:18:45 ERROR|   File "/usr/share/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/driver_load.py", line 29, in load_driver
04:18:45 ERROR|     status, output = session.cmd_status_output(cmd)
04:18:45 ERROR|   File "/usr/lib/python2.7/site-packages/aexpect/client.py", line 1136, in cmd_status_output
04:18:45 ERROR|     o = self.cmd_output(cmd, timeout, internal_timeout, print_func, safe)
04:18:45 ERROR|   File "/usr/lib/python2.7/site-packages/aexpect/client.py", line 1053, in cmd_output
04:18:45 ERROR|     o = self.read_up_to_prompt(timeout, internal_timeout, print_func)
04:18:45 ERROR|   File "/usr/lib/python2.7/site-packages/aexpect/client.py", line 1022, in read_up_to_prompt
04:18:45 ERROR|     print_func)[1]
04:18:45 ERROR|   File "/usr/lib/python2.7/site-packages/aexpect/client.py", line 853, in read_until_last_line_matches
04:18:45 ERROR|     print_func)
04:18:45 ERROR|   File "/usr/lib/python2.7/site-packages/aexpect/client.py", line 789, in read_until_output_matches
04:18:45 ERROR|     raise ExpectProcessTerminatedError(patterns, self.get_status(), o)
04:18:45 ERROR|   File "/usr/lib/python2.7/site-packages/aexpect/client.py", line 278, in get_status
04:18:45 ERROR|     wait_for_lock(self.lock_server_running_filename)
04:18:45 ERROR|   File "/usr/lib/python2.7/site-packages/aexpect/shared.py", line 37, in wait_for_lock
04:18:45 ERROR|     fd = get_lock_fd(filename)
04:18:45 ERROR|   File "/usr/lib/python2.7/site-packages/aexpect/shared.py", line 10, in get_lock_fd
04:18:45 ERROR|     open(filename, "w").close()
04:18:45 ERROR| IOError: [Errno 2] No such file or directory: '/tmp/aexpect_e1MRnTIL/lock-server-running'
04:18:45 ERROR|
04:18:45 ERROR| Traceback (most recent call last):

04:18:45 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-39.0-py2.7.egg/avocado_vt/test.py", line 434, in _run_avocado
    raise test_exception

04:18:45 ERROR| IOError: [Errno 2] No such file or directory: '/tmp/aexpect_e1MRnTIL/lock-server-running'

04:18:45 ERROR| ERROR 1-smp_2.16384m.repeat1.run_test.Host_RHEL.m7.u3.rng_random.qcow2.virtio_scsi.up.virtio_net.Win7.x86_64.sp1.io-github-autotest-qemu.driver_load_stress.during_bg_test.with_viorng -> IOError: [Errno 2] No such file or directory: '/tmp/aexpect_e1MRnTIL/lock-server-running'
04:18:45 INFO |

File: qemu/tests/driver_load.py [hunk 2]
```diff
@@ -70,24 +78,28 @@ def check_driver(session, cmd, pattern):
...
+            error_context.context("Unload the driver", logging.info)
+            unload_driver(session, driver_unload_cmd, driver_id)
+            session = vm.wait_for_login(nic_index=nic_index, timeout=timeout)
+            time.sleep(5)
+        except exceptions.TestFail:
+            raise
+        finally:
+            error_context.context("Load the driver", logging.info)
+            load_driver(session, driver_load_cmd, driver_id)
+            time.sleep(5)
```
Comment (CongLi) [inline qemu/tests/driver_load.py]: @suqinhuang 

According to the request, need run rng again after enable rng driver.
I'm not sure whether what's the final decision about it.

**Comment by suqinhuang** [comment]
PASS 1-smp_1.4096m.repeat1.run_test.Host_RHEL.m7.u2.virtio_rng.rng_random.qcow2.virtio_blk.up.virtio_net.Win8.x86_64.1.io-github-autotest-qemu.driver_load_stress.with_viorng.during_bg_test
PASS 1-smp_1.4096m.repeat1.run_test.Host_RHEL.m7.u2.virtio_rng.rng_random.qcow2.virtio_blk.up.virtio_net.Win8.x86_64.1.io-github-autotest-qemu.driver_load_stress.with_viorng.before_bg_test
PASS 2-smp_1.4096m.repeat1.run_test.Host_RHEL.m7.u2.virtio_rng.rng_random.qcow2.virtio_blk.up.virtio_net.Win8.x86_64.1.io-github-autotest-qemu.driver_load_stress.with_viorng.after_bg_test

**Comment by pingl2016** [comment]
@suqinhuang 
Test failed for case driver_load_stress.with_viorng.during_bg_test.
11:10:59 ERROR| Reproduced traceback from: /usr/lib/python2.7/site-packages/avocado_plugins_vt-42.0-py2.7.egg/avocado_vt/test.py:420
11:10:59 ERROR| Traceback (most recent call last):
11:10:59 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-42.0-py2.7.egg/avocado_vt/test.py", line 196, in runTest
11:10:59 ERROR|     self._runTest()
11:10:59 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-42.0-py2.7.egg/avocado_vt/test.py", line 337, in _runTest
11:10:59 ERROR|     run_func(self, params, env)
11:10:59 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-42.0-py2.7.egg/virttest/error_context.py", line 135, in new_fn
11:10:59 ERROR|     return fn(*args, **kwargs)
11:10:59 ERROR|   File "/usr/share/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/driver_in_use.py", line 110, in run
11:10:59 ERROR|     run_subtest(sub_type)
11:10:59 ERROR|   File "/usr/share/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/driver_in_use.py", line 80, in run_subtest
11:10:59 ERROR|     utils_test.run_virt_sub_test(test, params, env, sub_type)
11:10:59 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-42.0-py2.7.egg/virttest/utils_test/__init__.py", line 1242, in run_virt_sub_test
11:10:59 ERROR|     run_func(test, params, env)
11:10:59 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-42.0-py2.7.egg/virttest/error_context.py", line 135, in new_fn
11:10:59 ERROR|     return fn(*args, **kwargs)
11:10:59 ERROR|   File "/usr/share/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/driver_load.py", line 122, in run
11:10:59 ERROR|     sub_type=test_after_load)
11:10:59 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-42.0-py2.7.egg/virttest/utils_test/__init__.py", line 1242, in run_virt_sub_test
11:10:59 ERROR|     run_func(test, params, env)
11:10:59 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-42.0-py2.7.egg/virttest/error_context.py", line 135, in new_fn
11:10:59 ERROR|     return fn(*args, **kwargs)
11:10:59 ERROR|   File "/usr/share/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/rng_bat.py", line 130, in run
11:10:59 ERROR|     output = session.cmd_output(read_rng_cmd, timeout=read_rng_timeout)
11:10:59 ERROR|   File "/usr/lib/python2.7/site-packages/aexpect/client.py", line 1057, in cmd_output
11:10:59 ERROR|     raise ShellTimeoutError(cmd, o)
11:10:59 ERROR| ShellTimeoutError: Timeout expired while waiting for shell command to complete: 'for /l %i in (1, 1, 100) do "D:\\random_%PROCESSOR_ARCHITECTURE%.exe"'

File: qemu/tests/driver_load.py [hunk 3]
```diff
@@ -21,73 +22,99 @@ def run(test, params, env):
...
     driver_unload_cmd = params["driver_unload_cmd"]
 
-    devcon = params.get("devcon")
-    if devcon:
-        error.context("Copy devcon.exe from winutils.iso to C:\\")
+    if params.get("devcon"):
+        # Use the last nic for send driver load/unload command
+        nic_index = len(vm.virtnet) - 1
+        session = vm.wait_for_login(nic_index=nic_index, timeout=timeout)
+        error_context.context("Copy devcon.exe from winutils.iso to C:\\")
```
Comment (pingl2016) [inline qemu/tests/driver_load.py]: Shall we add the parameter "logging.info" to the function? As I cannot find the step description from the log.

**Comment by pingl2016** [comment]
Case failed as sub-test rng_bat can not be detected. Seem command "for /l %i in (1, 1, 20) do "D:\random_%PROCESSOR_ARCHITECTURE%.exe"" finished within 1 second.

4:13:13 ERROR| Reproduced traceback from: /usr/lib/python2.7/site-packages/avocado_plugins_vt-42.0-py2.7.egg/avocado_vt/test.py:420
04:13:13 ERROR| Traceback (most recent call last):
04:13:13 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-42.0-py2.7.egg/avocado_vt/test.py", line 196, in runTest
04:13:13 ERROR|     self._runTest()
04:13:13 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-42.0-py2.7.egg/avocado_vt/test.py", line 337, in _runTest
04:13:13 ERROR|     run_func(self, params, env)
04:13:13 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-42.0-py2.7.egg/virttest/error_context.py", line 135, in new_fn
04:13:13 ERROR|     return fn(*args, **kwargs)
04:13:13 ERROR|   File "/usr/share/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/driver_in_use.py", line 106, in run
04:13:13 ERROR|     stress_thread = run_bg_stress_test(bg_stress_test)
04:13:13 ERROR|   File "/usr/share/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/driver_in_use.py", line 64, in run_bg_stress_test
04:13:13 ERROR|     "alive!" % bg_stress_test)
04:13:13 ERROR| TestFail: Backgroud test rng_bat is not alive!
04:13:13 ERROR|
04:13:13 ERROR| FAIL 3-smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Win7.x86_64.sp1.io-github-autotest-qemu.driver_load_stress.with_viorng.during_bg_test -> TestFail: Backgroud test rng_bat is not alive!

**Comment by pingl2016** [comment]
Time to generate random number vary a lot on different guest os and host cpu. The detail info is as below.
Guest os: Win8.x86_64, host cpu: Intel. Take 1 second to generate random number 100 times.
Guest os: Win2012.x86_64, host cpu: Amd. Take 3600 seconds to generate random number 50 times.
Guest os: Win2012.x86_64.r2, host cpu: Amd. Take 3600 seconds to generate random number 65 times.
Guest os: Win8.x86_64.1, host cpu: Amd. Take 3600 seconds to generate random number 96 times.

**Comment by pingl2016** [comment]
Based on avocado-framework/avocado-vt#755

**Comment by pingl2016** [comment]
For intel host, command "for /l %i in (1, 1, 200) do X:\random_%PROCESSOR_ARCHITECTURE%.exe &" will be finished before unload and load virtio-rng.

File: qemu/tests/cfg/driver_load_stress.cfg [hunk 4]
```diff
@@ -4,17 +4,25 @@
...
     sub_test = driver_load
+    i386, i686:
+        driver_load_cmd = "WIN_UTILS:\devcon\wnet_x86\devcon.exe enable @DRIVER_ID"
+        driver_unload_cmd = "WIN_UTILS:\devcon\wnet_x86\devcon.exe disable @DRIVER_ID"
+        driver_id_cmd = "WIN_UTILS:\devcon\wnet_x86\devcon.exe find * | find 'VirtIO'"
+        driver_check_cmd = "WIN_UTILS:\devcon\wnet_x86\devcon.exe status @DRIVER_ID"
+    x86_64:
+        driver_load_cmd = "WIN_UTILS:\devcon\wnet_amd64\devcon.exe enable @DRIVER_ID"
+        driver_unload_cmd = "WIN_UTILS:\devcon\wnet_amd64\devcon.exe disable @DRIVER_ID"
+        driver_id_cmd = "WIN_UTILS:\devcon\wnet_amd64\devcon.exe find * | find 'VirtIO'"
```
Comment (pingl2016) [inline qemu/tests/cfg/driver_load_stress.cfg]: should be D:\devcon\wnet_x86\devcon.exe find * | find "VirtIO"

**Comment by suqinhuang** [comment]
https://github.com/autotest/tp-qemu/pull/813

**Comment by suqinhuang** [comment]
@pingl2016 updated, could you have a look, thanks

**Comment by pingl2016** [comment]
Acked-by: Ping Li <pingl@redhat.com>
 (1/3) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Win7.x86_64.sp1.io-github-autotest-qemu.driver_load_stress.with_viorng.before_bg_test: PASS (123.77 s)
 (2/3) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Win7.x86_64.sp1.io-github-autotest-qemu.driver_load_stress.with_viorng.during_bg_test: PASS (65.61 s)
 (3/3) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Win7.x86_64.sp1.io-github-autotest-qemu.driver_load_stress.with_viorng.after_bg_test: PASS (59.95 s)

**Comment by suqinhuang** [comment]
@xutian @luckyh could you help review this patch

File: qemu/tests/driver_load.py [hunk 5]
```diff
@@ -21,73 +22,94 @@ def run(test, params, env):
...
+        error_context.context("Unload and load the driver. Round %s" % repeat,
+                              logging.info)
+        logging.info("Get driver info from guest")
+        driver_id = get_driver_id(driver_id_cmd, driver_id_pattern)
+
+        try:
+            error_context.context("Unload the driver", logging.info)
+            unload_driver(driver_unload_cmd, driver_id)
+            time.sleep(5)
+        except exceptions.TestFail:
```
Comment (xutian) [inline qemu/tests/driver_load.py]: useless  try

Comment (suqinhuang) [inline qemu/tests/driver_load.py]: I'm afraid driver is unloaded which will effect the following test.
snapshot should resolve this issue.
code have been udpated, could you review again

**Comment by suqinhuang** [comment]
https://github.com/avocado-framework/avocado-vt/pull/757/files

**Comment by XueqiangWei** [comment]
LGTM,
Acked-by: XueqiangWei <xuwei@redhat.com>
(1/3) smp_2.2048m.repeat1.run_test.Host_RHEL.m7.u3.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Win7.x86_64.sp1.io-github-autotest-qemu.driver_load_stress.with_viorng.before_bg_test: PASS (184.71 s)
 (2/3) smp_2.2048m.repeat1.run_test.Host_RHEL.m7.u3.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Win7.x86_64.sp1.io-github-autotest-qemu.driver_load_stress.with_viorng.during_bg_test: PASS (249.76 s)
 (3/3) smp_2.2048m.repeat1.run_test.Host_RHEL.m7.u3.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.Win7.x86_64.sp1.io-github-autotest-qemu.driver_load_stress.with_viorng.after_bg_test: PASS (183.82 s)

**Comment by CongLi** [comment]
@suqinhuang @pingl2016 

How to solve this problem "04:18:45 ERROR| ERROR 1-smp_2.16384m.repeat1.run_test.Host_RHEL.m7.u3.rng_random.qcow2.virtio_scsi.up.virtio_net.Win7.x86_64.sp1.io-github-autotest-qemu.driver_load_stress.during_bg_test.with_viorng -> IOError: [Errno 2] No such file or directory: '/tmp/aexpect_e1MRnTIL/lock-server-running'" ?

**Comment by suqinhuang** [comment]
@CongLi it happen with latest code? it should because the using session is closed

**Comment by suqinhuang** [comment]
@pingl2016 @CongLi @xutian @XueqiangWei @luckyh 
I will merge it if no other comments, as it block several bugs

**Pull request merged**